test/core/client_channel/lb_policy/xds_override_host_test.cc

```diff
@@ -135,21 +131,96 @@ TEST_F(XdsOverrideHostTest, OverrideHostChannelNotFound) {
     subchannel->SetConnectivityState(GRPC_CHANNEL_CONNECTING);
     subchannel->SetConnectivityState(GRPC_CHANNEL_READY);
   }
-  ASSERT_NE(WaitForConnected(), nullptr);
-  ASSERT_NE(ExpectState(GRPC_CHANNEL_READY), nullptr);
-  auto picker = ExpectState(GRPC_CHANNEL_READY);
+  auto picker = WaitForConnected();
+  ASSERT_NE(picker, nullptr);
+  EXPECT_TRUE(ExpectPickComplete(picker.get()).has_value());
+  picker = ExpectState(GRPC_CHANNEL_READY);
+  ASSERT_NE(picker, nullptr);
+  EXPECT_TRUE(ExpectPickComplete(picker.get()).has_value());
+  picker = ExpectState(GRPC_CHANNEL_READY);
+  ASSERT_NE(picker, nullptr);
   // Make sure child policy works
   EXPECT_NE(ExpectPickComplete(picker.get()), ExpectPickComplete(picker.get()));
   // Check that the host is overridden
-  std::map<std::string, std::string> pick_arg{
-      {std::string(kOverrideHostHeaderName), "no such host"}};
-
+  std::map<UniqueTypeName, std::string> pick_arg{
+      {XdsHostOverrideTypeName(), "no such host"}};
   std::unordered_set<absl::optional<std::string>> picks{
       ExpectPickComplete(picker.get(), pick_arg),
       ExpectPickComplete(picker.get(), pick_arg)};
-
   ASSERT_THAT(picks, UnorderedElementsAre(kAddresses[0], kAddresses[1]));
 }
+
+TEST_F(XdsOverrideHostTest, DISABLED_SubchannelsComeAndGo) {
+  const std::array<absl::string_view, 3> kAddresses = {
+      "ipv4:127.0.0.1:441", "ipv4:127.0.0.1:442", "ipv4:127.0.0.1:443"};
+  EXPECT_EQ(ApplyUpdate(BuildUpdate(kAddresses,
+                                    MakeXdsOverrideHostConfig("round_robin")),
+                        policy_.get()),
+            absl::OkStatus());
+  ExpectConnectingUpdate();
+  RefCountedPtr<LoadBalancingPolicy::SubchannelPicker> picker;
+  // Ready up both subchannels
+  for (auto address : kAddresses) {
+    auto subchannel = FindSubchannel({address});
+    ASSERT_NE(subchannel, nullptr);
+    ASSERT_TRUE(subchannel->ConnectionRequested());
+    subchannel->SetConnectivityState(GRPC_CHANNEL_CONNECTING);
+    subchannel->SetConnectivityState(GRPC_CHANNEL_READY);
+    picker = WaitForConnected();
```

Comment:
I don't think it makes sense to check this inside the loop.  Let's check that we get connected just once, after we exit the loop.  After that, each subsequent update should be another READY update with a new picker.

Reply:
Done